sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -248,6 +258,75 @@ class wi_element {
                         PI_INVALID_DEVICE);
 #endif // __SYCL_DEVICE_ONLY__
   }
+  wi_element &
+  operator=(const wi_element<T, NumRows, NumCols, Layout, Group> &rhs) {
+#ifdef __SYCL_DEVICE_ONLY__
+    M.spvm = __spirv_VectorInsertDynamic(
+        M.spvm, __spirv_VectorExtractDynamic(rhs.M.spvm, rhs.idx), idx);
+    return *this;
+#else
+    (void)rhs;
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  friend T operator+(const wi_element<T, NumRows, NumCols, Layout, Group> &lhs,
+                     const T &rhs) {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx) + rhs;
+#else
+    (void)rhs;
```

Comment:
Is there a reason not to ignore `lhs` here and in other overloads with it?

Reply:
Nice catch, lhs should not be ignored